### PR TITLE
SIP-117: Support Synth Exchanging on L2 with Optimism

### PR DESCRIPTION
**Pull request by i-stam**

### CODE REVIEW

File: sips/sip-114.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+The polymorphic approach adopted during the L2 integration (check [SIP-102](./sip-102.md) for further details) proved to be very efficient. For basic exchanging functionality the functions needed to be added are the following: `exchange()`, `exchangeOnBehalf()`,`settle()`. Their implementation is moved from `Synthetix` to `BaseSynthetix` since they are going to be available on both layers. Due to the near-instant confirmation time of the transactions on L2, oracle front-running is no longer an issue, thus, exchanging needs no settlement. The waiting period parameter is set to 0 when deploying on L2 and there are extra checks in the code that bypass settlement to reduce unnecessary function and contract calls and hence gas costs. For example, no exchange entries are created via `appendExchange()` when the waiting period is 0.
+
+### Technical Specification
+
+<!--The technical specification should outline the public API of the changes proposed. That is, changes to any of the interfaces Synthetix currently exposes or the creations of new ones.-->
+
+```solidity
+interface IBaseDebtCache {
```
Comment (justinjmoses) [inline sips/sip-114.md:68]: I don't think we need to define this as we're sticking with the original interface

File: sips/sip-114.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+After a multi-phase, incentivised testnet trial, Synthetix went live on Optimistic Ethereum mainnet. However, this was an intitial soft launh i.e. L2 users can only stake SNX and issue or burn sUSD. The implementation of this SIP enables actual synth exchanging on L2.
+.
+
+## Motivation
+
+<!--This is the problem statement. This is the *why* of the SIP. It should clearly explain *why* the current state of the protocol is inadequate.  It is critical that you explain *why* the change is needed, if the SIP proposes changing how something is calculated, you must address *why* the current calculation is innaccurate or wrong. This is not the place to describe how the SIP will address the issue!-->
+
+As described in [The Optimistic Ethereum Transition](https://blog.synthetix.io/the-optimistic-ethereum-transition), phase 0 enabled users to lock snx on and mint an equivalent amount via a Syntehtix to Optimism bridge. However, the L2 functionality is limited to staking SNX, issuing sUSD and claiming weekly rewards. Additionally, both SNX and sUSD are not tarnsferable, due to an initial security related restriction imposed by OE.
```
Comment (justinjmoses) [inline sips/sip-114.md:27]: Can you mention somewhere that this is Phase 1?

File: sips/sip-114.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+<!--"If you can't explain it simply, you don't understand it well enough." Simply describe the outcome the proposed changes intends to achieve. This should be non-technical and accessible to a casual community member.-->
+
+Enable synth exchanging on L2 via Optimism
+
+## Abstract
+
+<!--A short (~200 word) description of the proposed change, the abstract should clearly describe the proposed change. This is what *will* be done if the SIP is implemented, not *why* it should be done or *how* it will be done. If the SIP proposes deploying a new contract, write, "we propose to deploy a new contract that will do x".-->
+
+After a multi-phase, incentivised testnet trial, Synthetix went live on Optimistic Ethereum mainnet. However, this was an intitial soft launh i.e. L2 users can only stake SNX and issue or burn sUSD. The implementation of this SIP enables actual synth exchanging on L2.
```
Comment (justinjmoses) [inline sips/sip-114.md:20]: `s/launh/launch/`

File: sips/sip-114.md (hunk 2 above)
Comment (justinjmoses) [reply sips/sip-114.md:27]: `s/tarnsferable/transferable`

File: sips/sip-114.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+4. Test Cases
+5. Configurable Values
+-->
+
+### Overview
+
+<!--  -->
+<!--This is a high level overview of *how* the SIP will solve the problem. The overview should clearly describe how the new feature will be implemented.-->
+
+No new contracts were developed. The OE Synthetix instance, namely `MintableSynthetix` was enhanced with exchanging functionality. The basic Synhetix functionality that is supoorted on both layers is imlemented on a parent contract (`BaseSynthetix`) which is inherited by the L1 (Synthetix) and the l2 (MintableSynthetix) instances.
```
Comment (justinjmoses) [inline sips/sip-114.md:45]: `s/Synhetix/Synthetix`

Comment (justinjmoses) [reply sips/sip-114.md:45]: `s/imlemented/implemented`

File: sips/sip-114.md [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+> Note: A `BaseDebtCahce` contract was introduced in order to state the shared functionality between DebtCahce and RealtimedebtCache more explicitly.
+
+The waiting period is now set to 0 and no rebate/reclaim takes place.
```
Comment (justinjmoses) [inline sips/sip-114.md:49]: This should be removed and added into another SIP

File: sips/sip-114.md [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+
+The following functions were activated on L2 `Synthetix`:
```
Comment (justinjmoses) [inline sips/sip-114.md:51]: *Are to be activated*

File: sips/sip-114.md [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+
+The polymorphic approach adopted during the L2 integration (check [SIP-102](./sip-102.md) for further details) proved to be very efficient. For basic exchanging functionality the functions needed to be added are the following: `exchange()`, `exchangeOnBehalf()`,`settle()`. Their implementation is moved from `Synthetix` to `BaseSynthetix` since they are going to be available on both layers. Due to the near-instant confirmation time of the transactions on L2, oracle front-running is no longer an issue, thus, exchanging needs no settlement. The waiting period parameter is set to 0 when deploying on L2 and there are extra checks in the code that bypass settlement to reduce unnecessary function and contract calls and hence gas costs. For example, no exchange entries are created via `appendExchange()` when the waiting period is 0.
```
Comment (justinjmoses) [inline sips/sip-114.md:61]: Please remove the part on the waiting period here as well

**Review by justinjmoses — CHANGES_REQUESTED** [review]
Changes inline.

**Review by ajsantander — CHANGES_REQUESTED** [review]
Looks good! Suggested minor wording changes in https://github.com/Synthetixio/SIPs/pull/404

File: sips/sip-117.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
+---
+sip: 117
+title: Support Synth Exchanging on L2 with Optimism
+status: Proposed
+author:
```
Comment (ajsantander) [inline sips/sip-117.md:5]: This section is incomplete

**Review by justinjmoses — APPROVED** [review]

**Review by justinjmoses — APPROVED** [review]
Please just create and add the discussions thread

**Review by justinjmoses — APPROVED** [review]

**Pull request merged**